gui/guihtml/src/TGHtml.cxx

```diff
@@ -1365,6 +1366,17 @@ Bool_t TGHtml::HandleButton(Event_t *event)
             //!!delete[] uri;
          }
       }
+   } else if ((event->fType == kButtonPress) && (event->fCode == kButton3)) {
+      TGFileInfo fi;
+      const char *inputFileTypes[] = {
+          "HTML files",   "*.html",
+                     0,          0 };
+      fi.fFileTypes = inputFileTypes;
+      new TGFileDialog(gClient->GetRoot(), this, kFDSave, &fi);
```

Comment:
Unmatched `delete`.

Reply:
TGFileDialog gets automatically destructed when closing the window.
![image](https://user-images.githubusercontent.com/10653970/109956653-e4866280-7cb1-11eb-96d2-9578f39eea41.png)

If I add a delete later on, I get a segfault.

Reply:
See for example:
https://github.com/root-project/root/blob/f8efb11a51cbe5b5152ebef19a4f7b78744ca2fa/gui/guihtml/src/TGHtmlBrowser.cxx#L627

Reply:
I have seen that `DeleteWindow()` defers a call to `ReallyDelete()`, which boils down to this thing:
```
delete this;
```

Essentially, this breaks standard resource management in C++, e.g. `std::shared_ptr<...>` that do not provide a custom deleter.  Stack-allocated objects will also cause the program to crash.  In any case, these changes are out of the scope of this PR.

Reply:
Yes, I hate that too, hahaha.

ROOT drives me crazy with all these tricks, but after some years I learned how to cope with it. But looking forward to a safer ROOT in the future :)